### PR TITLE
Script Generate Azcopy Throughput Report

### PR DESCRIPTION
**Pull request by tilimpo**
- Analyzes the azcopy log, and collects transfer metrics to generate a repot Example: .\AzcopyThroughput.ps1 -F "C:\Users\tml\.azcopy\f2435d64-57ea-6949-7e62-006aa4ecf930.log"

### CODE REVIEW

**Comment by vibhansa-msft** [comment]
Kindly push these files to a seperate 'tools' or 'utils' directory instead of root.

File: utils/AzcopyThroughput.ps1 [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+if (Test-Path -Path $folder) {}else{
+    New-item -Path "C:\temp" -ItemType Directory
+}
+# Import the azcopy log
+Write-Host "`n Loading Azcopy log ($path)......" -ForegroundColor Yellow
+
+$timer=[system.diagnostics.stopwatch]::StartNew()
+$logname = (Get-ChildItem $path).BaseName
+
+# Read log, and save into auxiliar file, with all the information about azcoy metrics
```
Comment (gapra-msft) [inline utils/AzcopyThroughput.ps1:68]:
```suggestion
# Read log, and save into auxiliary file, with all the information about azcoy metrics

```

File: utils/AzcopyThroughput.ps1 [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+if (Test-Path -Path $folder) {}else{
+    New-item -Path "C:\temp" -ItemType Directory
+}
+# Import the azcopy log
+Write-Host "`n Loading Azcopy log ($path)......" -ForegroundColor Yellow
+
+$timer=[system.diagnostics.stopwatch]::StartNew()
+$logname = (Get-ChildItem $path).BaseName
+
+# Read log, and save into auxiliary file, with all the information about azcoy metrics
```
Comment (gapra-msft) [inline utils/AzcopyThroughput.ps1:68]:
```suggestion
# Read log, and save into auxiliary file, with all the information about azcopy metrics

```

File: utils/AzcopyThroughput.ps1 [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    }elseif($line -match 'Total Number of Transfers'){
+        $totalfiles = $line
+        break
+    }
+}
+$reader.Dispose()
+# Get the total files to transfer
+$tf=$totalfiles -split ':', 2
+
+# Convert the auxiliar file in a CSV to be imported in HTML report
```
Comment (gapra-msft) [inline utils/AzcopyThroughput.ps1:85]:
```suggestion
# Convert the auxiliayr file in a CSV to be imported in HTML report

```

Comment (gapra-msft) [inline utils/AzcopyThroughput.ps1:85]:
```suggestion
# Convert the auxiliary file in a CSV to be imported in HTML report

```

**Review by gapra-msft — COMMENTED** [review]
Hi, there seems to be quite a few typos in this file, particularly with the spelling of AzCopy, auxiliary, and report. If you could please resolve those, that would be super helpful!

Otherwise looks good. Will approve once those changes are made

**Comment by tilimpo** [comment]
> Hi, there seems to be quite a few typos in this file, particularly with the spelling of AzCopy, auxiliary, and report. If you could please resolve those, that would be super helpful!
> 
> Otherwise looks good. Will approve once those changes are made

Hi @gapra-msft, typos already corrected.

Thank you

**Review by gapra-msft — APPROVED** [review]

**Review by adreed-msft — COMMENTED** [review]
I would greatly prefer that the plotly.js script were not integrated into the template HTML file itself, but instead used the script reference provided on the plotly.js github page.

https://github.com/plotly/plotly.js/#the-script-html-element

This comes with a couple of distinct advantages:
1) It is trivial to update in the future
2) It is from a reliable trustworthy source (Not that I don't trust you, but a certain level of skepticism has to be taken to a good few thousand lines of minified js)
3) It doesn't prevent the file from being reviewed on GitHub by making it too massive

In addition to this, we need to add a notice to our licenses for the new use of a third-party component (LICENSES.txt in the root).

**Review by adreed-msft — APPROVED** [review]

**Comment by tilimpo** [comment]
> ng Contributor License Agreement(CLA). If you agree with the CLA, please reply with the follow

@microsoft-github-policy-service agree company="Microsoft"

**Comment by tilimpo** [comment]
[ReportHTML.zip](https://github.com/Azure/azure-storage-azcopy/files/12844143/ReportHTML.zip)
I made some changes to the HTML file, which I would like to replace the existing one, the changes are as follows:
- replacement of the js code with the public plotly link
- addition of warnings to the graph
- addition of a table with findings, which shows all the warnings that we had during the copy and what the meaning of these same warnings is.

**Pull request closed without merging**